### PR TITLE
Use custom DebugProxyHost to initialize DebugProxy config

### PR DESCRIPTION
**Pull request by captainsafia**
This PR creates a new DebugProxyHost that is used to configure and initialize the debugging proxy.

I migrated some of the applicable configs from the WebHostBuilder and migrated our configs that were in Program.cs to this class.

Following @pranavkm I also added support for loading configurations from a JSON file should we need this in the future.

Addresses #19909

### CODE REVIEW

**Review by SteveSandersonMS — APPROVED** [review]
This looks way better. Awesome!

**Comment by SteveSandersonMS** [comment]
Oh BTW if you're merging this today, I think it should go into the `blazor-wasm-preview4` branch (not `blazor-wasm`) unless @mkArtakMSFT still thinks we should take preview3 changes. I think it would be fine to have this in preview4, but am fine either way.

File: src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            })
+            .ConfigureWebHostDefaults(webBuilder =>
+            {
+                webBuilder.UseStartup<Startup>();
+
+                // By default we bind to a dyamic port
+                // This can be overridden using an option like "--urls http://localhost:9500"
+                webBuilder.UseUrls("http://127.0.0.1:0");
+            })
+            .UseDefaultServiceProvider((context, options) =>
```
Comment (pranavkm) [inline src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs]: Might be ok to remove this. Scope validation is useful if you want to verify if the DI was mis-configured. We don't need that here.

File: src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
+using System;
```
Comment (pranavkm) [inline src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs]:
```suggestion
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

using System;
```

**Review by pranavkm — APPROVED** [review]

File: src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs (hunk 1 above)
Comment (captainsafia) [reply src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs]: Done.

File: src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs (hunk 2 above)
Comment (captainsafia) [reply src/Components/WebAssembly/DebugProxy/src/Hosting/DebugProxyHost.cs]: ❗️ Done.

**Pull request merged**